### PR TITLE
RSDK-1967 Genericize FromRobot

### PR DESCRIPTION
**Pull request by oliviamiller**
Created method ResourceFromRobot in robot.go to use in FromRobot.

### CODE REVIEW

File: robot/robot.go [hunk 1]
```diff
@@ -225,3 +230,18 @@ func ResourceFromProtoMessage(
...
+	if err != nil {
+		return nil, err
+	}
+
+	part, ok := res.(T)
+
+	if !ok {
+		return nil, NewUnimplementedInterfaceError[T](res)
+	}
+	return &part, nil
```
[the remark this answers is not in the document]
Comment (oliviamiller) [reply robot/robot.go:246]: yes, otherwise we can't return nil when there's an error @cheukt

[the remark this answers is not in the document]
Comment (edaniels) [reply robot/robot.go:246]: you can use `var zero T` as a means to return the zero type of a T, which may be nil.

[the remark this answers is not in the document]
Comment (oliviamiller) [reply robot/robot.go]: done, thanks!

[the remark this answers is not in the document]
Comment (maximpertsov) [reply robot/robot.go]: cool didn't know about this - would the return type still need to be a pointer if we use `var zero T` ?

[the remark this answers is not in the document]
Comment (edaniels) [reply robot/robot.go]: if the T coming in is a *ConcreteT, then it's already that. It depends on the input.

**Review by maximpertsov — APPROVED** [review]

**Review by cheukt — APPROVED** [review]
LGTM

**Pull request merged**